prompt2model/param_selector/search_with_optuna.py

```diff
@@ -0,0 +1,188 @@
+"""This module provides a dummy trainer for testing purposes."""
+from __future__ import annotations  # noqa FI58
+
+from typing import Any
+
+import transformers
+from optuna.trial import Trial
+
+from datasets import Dataset
+from transformers import PreTrainedModel  # noqa
+from transformers import PreTrainedTokenizer
+from prompt2model.prompt_parser.base import PromptSpec
+
+from prompt2model.utils.config import DEFAULT_HYPERPARAMETERS, DEFAULT_HYPERPARAMETERS_SPACE
+from prompt2model.model_trainer import BaseTrainer
+from prompt2model.param_selector.base import ParamSelector
+
+# TODO:
+# - User tweaking hyperparameter
+# - Dynamic initialization of hyperparamter range from task type and complexity
+# - Using LLM to suggest hyperparameter
+
+
+class OptunaParamSelector(ParamSelector):
+    """Uses optuna for searching for hyperparameters"""
+
+    def __init__(self, trainer: BaseTrainer):
```

Comment:
Suggestion: rename this object to `promptmodel_trainer` to avoid clashing with the huggingface trainer member variable name within BaseTrainer. 
```suggestion
    def __init__(self, promptmodel_trainer: BaseTrainer):
```